addon/MetadataStore.js

```diff
@@ -525,6 +525,39 @@ MetadataStore.prototype = {
   }),
 
   /**
+   * Find the oldest entry in the database
+   *
+   * Returns the timestamp of the oldest entry in the database
+   */
+  asyncGetOldestInsert: Task.async(function*() {
+    let timestamp = null;
+    try {
+      const entry = yield this.asyncExecuteQuery("SELECT created_at FROM page_metadata ORDER BY created_at ASC LIMIT 1");
+      if (entry.length) {
+        timestamp = Date.parse(entry[0][0]);
+      }
+    } catch (e) {
+      Cu.reportError(e);
```

Comment:
I recall that we used to have a PR to prevent MetadataStore from logging those lengthy errors/exceptions, perhaps just keep silent here, and simply propagate the exception to the client instead.